src/benchmarks/micro/libraries/System.Formats.Tar/Perf.TarFile.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+
+namespace System.Formats.Tar.Tests
+{
+    [BenchmarkCategory(Categories.Libraries)]
+    public class Perf_TarFile
+    {
+        /*
+        Always exist:
+            root/
+                inputdir/
+                        testdir/
+                        file.txt
+
+        Created on demand:
+                outputdir/
+                output.tar
+        */
+        private static readonly string _rootDirPath = FileUtils.GetTestFilePath();
+        private static readonly string _inputDirPath = Path.Combine(_rootDirPath, "inputdir");
+        private static readonly string _outputDirPath = Path.Combine(_rootDirPath, "outputdir");
+        private static readonly string _testDirPath = Path.Combine(_inputDirPath, "testdir");
+        private static readonly string _testFilePath = Path.Combine(_inputDirPath, "file.txt");
+        private static readonly string _inputTarFilePath = Path.Combine(_rootDirPath, "input.tar");
+        private static readonly string _outputTarFilePath = Path.Combine(_rootDirPath, "output.tar");
+        private MemoryStream _memoryStream = null;
+
+
+        // Setup and Cleanup
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            Directory.CreateDirectory(_rootDirPath);
+            Directory.CreateDirectory(_inputDirPath);
+            Directory.CreateDirectory(_testDirPath);
+            File.Create(_testFilePath).Dispose();
+            TarFile.CreateFromDirectory(sourceDirectoryName: _inputDirPath, destinationFileName: _inputTarFilePath, includeBaseDirectory: false);
+        }
+
+        [GlobalCleanup]
+        public void Cleanup()
+        {
+            Directory.Delete(_rootDirPath, recursive: true);
+            CleanupMemoryStream();
+        }
+        
+        [IterationSetup]
+        public void SetupIteration()
+        {
+            if (File.Exists(_outputTarFilePath))
+            {
+                File.Delete(_outputTarFilePath);
+            }
+            if (Directory.Exists(_outputDirPath))
+            {
+                Directory.Delete(_outputDirPath, recursive: true);
+            }
+            Directory.CreateDirectory(_outputDirPath);
+            CleanupMemoryStream();
+            _memoryStream = new MemoryStream();
```

Comment:
Yeah I saw the comment for `IterationSetup` to avoid using it for microbenchmarks, but I also noticed we use it _everywhere_ in microbenchmarks :D so I wanted to get feedback here, in case that documentation was obsolete somehow.

Why does `IterationSetup` make the results less stable? Is it because the performance of those methods is counted towards the final report instead of skipped?

Reply:
> Why does IterationSetup make the results less stable?

It's because when `IterationSetup` is used, BDN performs a single invocation per iteration.

So it's sth like:

```cs
IterationSetup();
Stopwatch.Start();
Benchmark();
Stopwatch.Stop();
```

while without `IterationSetup`, it performs multiple operations:

```cs
Stopwatch.Start();
for (long i = 0; i < invocationCount; i++)
   Benchmark();
Stopwatch.Stop();
```

This heavily affects [The Central Limit Theorem](https://learning.oreilly.com/library/view/pro-net-benchmarking/9781484249413/html/437795_1_En_4_Chapter.xhtml#:-:text=The%20Central%20Limit%20Theorem) which BDN relies on. And `Stopwatch` resolution is limited, so for nano-benchmarks, the difference between start and stop timestamps can be equal zero.

Reply:
I got rid of IterationSetup, I used your suggestions.